Core/src/org/sleuthkit/autopsy/guiutils/WrapLayout.java

```diff
@@ -0,0 +1,607 @@
+ * be aligned to the right.
+ *
+ * Inspired by WrapLayout
+ * https://tips4java.wordpress.com/2008/11/06/wrap-layout/ and FlowLayout
+ * https://raw.githubusercontent.com/mynawang/Java8-Source-Code/master/src/main/jdk8/java/awt/FlowLayout.java.
+ */
+public class WrapLayout implements LayoutManager, java.io.Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * The <code>WrapLayout</code> manager allows a seperation of components
+     * with gaps. The horizontal gap will specify the space between components
+     * and between the components and the borders of the <code>Container</code>.
+     *
+     * @serial
+     * @see #getHgap()
+     * @see #setHgap(int)
+     */
+    private int hgap = 0;
+
+    /**
+     * The <code>WrapLayout</code> manager allows a seperation of components
+     * with gaps. The vertical gap will specify the space between rows and
+     * between the the rows and the borders of the <code>Container</code>.
+     *
+     * @serial
+     * @see #getVgap()
+     * @see #setVgap(int)
+     */
+    private int vgap = 0;
+
+    /**
+     * If true, components will be aligned on their baseline.
+     */
+    private boolean alignOnBaseline = false;
+
+    /**
+     * The set of components that will be aligned on the opposite side (if left
+     * to right, on the right).
+     */
+    private final Set<Component> oppositeAlignedItems = new HashSet<>();
+
+    /**
+     * Constructs a new <code>WrapLayout</code> with a left alignment and a
+     * default 5-unit horizontal and vertical gap.
+     */
+    public WrapLayout() {
+        this(5, 5);
+    }
+
+    /**
+     * Constructs a new <code>WrapLayout</code> with a left alignment.
+     * 
+     * @param vgap  The vertical gap spacing between rows of components.
+     * @param hgap  The horizontal gap spacing between components.
+     */
+    public WrapLayout(int vgap, int hgap) {
+        this.vgap = vgap;
+        this.hgap = hgap;
+    }
+
+    /**
+     * Items in the collection will be aligned opposite to the rest. For
+     * instance, if items should be displayed left to right based on locale,
+     * these components will be on the right.
+     *
+     * @param oppAlignedComponents The components to display with opposite
+     *                             alignment.
+     */
+    public void setOppositeAligned(Collection<Component> oppAlignedComponents) {
+        synchronized (this.oppositeAlignedItems) {
+            this.oppositeAlignedItems.clear();
+            this.oppositeAlignedItems.addAll(oppAlignedComponents);
+        }
+    }
+
+    /**
+     * Items in the collection will be aligned opposite to the rest. For
+     * instance, if items should be displayed left to right based on locale,
+     * these components will be on the right.
+     *
+     * @return The components to display with opposite alignment.
+     */
+    public Collection<Component> getOppositeAlignedItems() {
+        return oppositeAlignedItems;
```

Comment:
return of collection field.

Reply:
Do you have a preference how this is handled (i.e. returning a new list generated from the original or something else)?

Reply:
Generally we have been using Collections.unmodifiableCollection(...)